src/devices/cardin.c

```diff
@@ -92,40 +102,56 @@ static int cardin_callback(bitbuffer_t *bitbuffer) {
 				dip[8]='+';
 		}
 
-		fprintf(stdout, "                 123456789\n");
-		fprintf(stdout, "dipswitch      = %s\n\n",dip);
-
-		fprintf(stdout, "                 -->ON\n");
-		fprintf(stdout, "right button   = ");
+		// Jumpers for the right button
 		if((bb[0][2] & 3) == 3) {
-			fprintf(stdout,                  "2 --o (this is right button)\n");
-			fprintf(stdout, "                 1 --o\n");
+			strcpy(rbuttonJ2, "--o (this is right button)");
+			strcpy(rbuttonJ1, "--o");
 		}
 		if((bb[0][2] & 9) == 9) {
-			fprintf(stdout,                  "2 --o (this is right button)\n");
-			fprintf(stdout, "                 1 o--\n");
+			strcpy(rbuttonJ2, "--o (this is right button)");
+			strcpy(rbuttonJ1, "o--");
 		}
 		if((bb[0][2] & 12) == 12) {
-			fprintf(stdout,                  "2 o-- (this is left button or two buttons on same channel)\n");
-			fprintf(stdout, "                 1 o--\n");
+			strcpy(rbuttonJ2, "o-- (this is left button or two buttons on same channel)");
```

Comment:
DIP1 is tristate
DIP2 is not

I did so : 
keeping "+-+-+-+-o" style for the main dipswitch (DIP1)
using "11R", "10R", "01R" and "00L?"
getting ride of strcpy() and the tests on bb[0][2]